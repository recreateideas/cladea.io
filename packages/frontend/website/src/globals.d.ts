import { Theme } from "./App/ui-core/theme";
import { Store } from "redux";

export declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: any;
    /** enzyme global methods for testing */
    shallow: any;
    render: any;
    mount: any;
    theme: Theme;
    ResizeObserver: any;
    store: Store;
  }
}

declare module "ui-core";
