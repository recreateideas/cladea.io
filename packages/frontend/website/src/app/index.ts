export * from "./App";
export * from "./features";
