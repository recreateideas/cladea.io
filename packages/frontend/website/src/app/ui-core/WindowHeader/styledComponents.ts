import styled from 'styled-components';

export const Container = styled.div`
    -webkit-app-region: drag;
    cursor: default;
    display: flex;
`;

export const Button = styled.div`
    /* -webkit-app-region: no-drag; */
    font-size: 9pt;
    line-height: 11px;
    margin-left: 4px;
    margin-right: 3px;
    width: 11px;
    height: 11px;
    line-height: 10px;
    border-radius: 50%;
    display: inline-block;
    display: flex;
    justify-content: center;
    div {
        visibility: hidden;
    }
    color: black;
    &.min-btn {
        background: #ffbd4c;
        border: 1px solid #e09e3e;
        span {
            font-size: 19px;
        }
    }
    &.max-btn {
        background: #00ca56;
        border: 1px solid #00ca56;
        font-size: 10px;
    }
    &.close-btn {
        background: #ff5c5c;
        border: 1px solid #ff5c5c;
        span {
            font-size: 10px;
            top: 5px;
            position: absolute;
        }
    }
`;

export const SidebarSection = styled.div`
    -webkit-app-region: drag;
    min-width: 199px;
    background-color: ${(props) => props.theme.dsl.palette.primaryBg};
`;

export const PageSection = styled.div`
    width: 100%;
    -webkit-app-region: drag;
`;

export const DotsButtons = styled.div`
    -webkit-app-region: drag;
    display: flex;
    padding: 5px 3px 0px 3px;
    box-sizing: border-box;
    width: fit-content;
    &:hover div {
        visibility: visible;
    }
`;
