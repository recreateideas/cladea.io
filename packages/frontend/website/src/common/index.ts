export * from './utils';
export * from './hooks';
export * from './converters';
