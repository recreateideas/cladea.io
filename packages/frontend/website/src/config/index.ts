export const apiGatewaUrl = process.env.REACT_APP_API_GATEWAY_URL;

console.log(process.env);
