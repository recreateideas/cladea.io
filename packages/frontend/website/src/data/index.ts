import land110m from "./land-110m.json";

export { land110m };
