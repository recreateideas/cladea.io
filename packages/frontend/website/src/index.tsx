import "./app/ui-core/theme/GlobalStyle/roboto.css";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import { store } from "./redux";
import { ServicedApp, ThemeProviderConnected } from "./app";

ReactDOM.render(
  <Provider store={store}>
    <ThemeProviderConnected>
      <ServicedApp />
    </ThemeProviderConnected>
  </Provider>,
  document.getElementById("root")
);
