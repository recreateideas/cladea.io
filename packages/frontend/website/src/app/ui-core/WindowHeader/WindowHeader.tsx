import { Container, Button, SidebarSection, PageSection, DotsButtons } from './styledComponents';
const { BrowserWindow } = window.require('electron').remote;

interface IProps {
    title?: string;
}
export const WindowHeader = (props: IProps) => {
    const { title = '' } = props;
    const minimize = () => {
        BrowserWindow.getFocusedWindow()?.minimize();
    };
    const maximize = () => {
        BrowserWindow.getFocusedWindow()?.maximize();
    };
    const close = () => {
        BrowserWindow.getFocusedWindow()?.close();
    };
    return (
        <Container className="window-header">
            <SidebarSection>
                <DotsButtons className="title-bar-btns">
                    <Button className="close-btn" onClick={close}>
                        <div>x</div>
                    </Button>
                    <Button className="min-btn" onClick={minimize}>
                        <div>-</div>
                    </Button>
                    <Button className="max-btn" onClick={maximize}>
                        <div>+</div>
                    </Button>
                </DotsButtons>
            </SidebarSection>
            <PageSection>
                <div className="title">{title}</div>
            </PageSection>
        </Container>
    );
};
