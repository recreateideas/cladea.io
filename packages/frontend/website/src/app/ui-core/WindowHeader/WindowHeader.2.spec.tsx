import { WindowHeader } from './WindowHeader';
import { TestProvider } from 'src/tests';

const { shallow } = TestProvider({});

describe('<WindowHeader/>', () => {
    it('should render', () => {
        const root = shallow(<WindowHeader />);
        expect(root).toMatchSnapshot();
    });
});
