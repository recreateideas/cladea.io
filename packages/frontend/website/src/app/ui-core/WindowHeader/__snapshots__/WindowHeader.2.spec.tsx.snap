// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WindowHeader/> should render 1`] = `
<styled.div
  className="window-header"
>
  <styled.div>
    <styled.div
      className="title-bar-btns"
    >
      <styled.div
        className="close-btn"
        onClick={[Function]}
      >
        <div>
          x
        </div>
      </styled.div>
      <styled.div
        className="min-btn"
        onClick={[Function]}
      >
        <div>
          -
        </div>
      </styled.div>
      <styled.div
        className="max-btn"
        onClick={[Function]}
      >
        <div>
          +
        </div>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    <div
      className="title"
    />
  </styled.div>
</styled.div>
`;
